pkg/kubelet/kubelet.go

```diff
@@ -2366,6 +2371,9 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 		if !networkConfigured {
 			messages = append(reasons, "network not configured correctly")
 		}
+		if !selinuxConfigured {
+			messages = append(reasons, "SELinux is required by the cluster but not enabled on this node")
```

Comment:
@dchen1107 

> * Admins have to decide if a cluster with SELinux or not at the cluster creation time. If they want a different configuration, they have to create a new one.

Not quite.  Technically I think it would be possible to turn SELinux integration **on** -- and the system will DTRT so that containers correctly with SELinux
enabled but are not isolated from one another (ie, they have a usable SELinux context for the system policy but no MCS labels).

That said, I think it is not required at all to work out the state transitions.  I don't know of anyone that wants to toggle back and forth between enabled and disabled.
If you run into a policy issue, you can setenforce 0 on the node and SELinux will allow actions that are denied by policy.  Since supporting toggling between SELinux enabled and disabled isn't a priority use-case at all, I'm fine defining the semantics as you suggest.

> * If master is configured with SELinux, all Kubelets should configured with SELinux; any Kubelet without such config should be rejected by master by marking as NotReady.
> * If master is configured withoutLinux, all Kubelets shouldn't configured with SELinux; any Kubelet with such config should be rejected by master by marking as NotReady. 

Yes

> In this case, why we need an extra flag in Kubelet here? Why not introducing such to NodeSpec?

I'm not sure what you mean by an extra flag in the Kubelet.  Elaborate?  And, will NodeSpec cover the nil cloud provider case?